### PR TITLE
feat: update create-next-app to init with main as initial branch

### LINKED ISSUE #17733: Change default branch for create-next-app output to be `main`
# Feature request

## Is your feature request related to a problem? Please describe.

Currently, the generated codebase from `npx create-next-app x` yields a project named `x` with git initialized with `master` as the default branch. To match up with GitHub, it should be `main`.

### PR DESCRIPTION
**Pull request by sidthesloth92**
When `create-next-app` is initialized a git repo, it now uses 'main' as the initial branch.

The branch master does not actually exist initially. The branches get created only when they have at least one commit. Until the branch gets created, the branch only exists in .git/HEAD. So there is no master branch initialized in the repo.

Closes: https://github.com/vercel/next.js/issues/17733

Screenshot 2020-10-09 at 17 26 30

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| buildDuration | 14.4s | 15.8s | ⚠️ +1.4s |
| nodeModulesSize | 63.4 MB | 63.4 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.799 | 2.765 | -0.03 |
| / avg req/sec | 893.28 | 904.18 | +10.9 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.608 | 1.613 | 0 |
| /error-in-render avg req/sec | 1554.44 | 1549.61 | ⚠️ -4.83 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..9b19.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-ef5fcfe..d66c.js gzip | 7.24 kB | 7.24 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58 kB | 58 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.9 kB | 6.9 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-1c7f28c..dule.js gzip | 6.29 kB | 6.29 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.9 kB | 52.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-409b283..e3ab.js gzip | 1.32 kB | 1.32 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.73 kB | 7.73 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-92d3016..dule.js gzip | 1.28 kB | 1.28 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 995 B | 995 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| buildDuration | 16.8s | 16.3s | -416ms |
| nodeModulesSize | 63.4 MB | 63.4 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..9b19.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-ef5fcfe..d66c.js gzip | 7.24 kB | 7.24 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58 kB | 58 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.9 kB | 6.9 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-1c7f28c..dule.js gzip | 6.29 kB | 6.29 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.9 kB | 52.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-409b283..e3ab.js gzip | 1.32 kB | 1.32 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.73 kB | 7.73 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-92d3016..dule.js gzip | 1.28 kB | 1.28 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _error.js | 1.05 MB | 1.05 MB | ✓ |
| 404.html | 4.34 kB | 4.34 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.05 MB | 1.05 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.09 MB | 1.09 MB | ✓ |
| withRouter.js | 1.09 MB | 1.09 MB | ✓ |
| Overall change | 5.41 MB | 5.41 MB | ✓ |

</details>

</details>
Commit: dbe565e0f5b0c18ffbe53ab90795924bad0b15af

File: packages/create-next-app/helpers/git.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ export function tryGitInit(root: string): boolean {
     execSync('git init', { stdio: 'ignore' })
     didInit = true
 
+    execSync('git checkout -b main', { stdio: 'ignore' })
```
Comment (asumaran) [inline packages/create-next-app/helpers/git.ts:33]: by doing this the repo will end up having two branches right? why not just rename the branch?

```
git branch -m main
```

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Thanks! This could still use an integration test though, but looks fine to merge.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| buildDuration | 14.5s | 14.4s | -29ms |
| nodeModulesSize | 84.9 MB | 84.9 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.627 | 2.683 | ⚠️ +0.06 |
| / avg req/sec | 951.59 | 931.74 | ⚠️ -19.85 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.43 | 1.484 | ⚠️ +0.05 |
| /error-in-render avg req/sec | 1748.82 | 1684.13 | ⚠️ -64.69 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d942eb5..343f.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-34e64ff..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 995 B | 995 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| buildDuration | 16.3s | 15.7s | -524ms |
| nodeModulesSize | 84.9 MB | 84.9 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d942eb5..343f.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-34e64ff..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | sidthesloth92/next.js feature/change-initial-branch-to-main | Change |
| - | - | - | - |
| _error.js | 915 kB | 915 kB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 915 kB | 915 kB | ✓ |
| link.js | 964 kB | 964 kB | ✓ |
| routerDirect.js | 957 kB | 957 kB | ✓ |
| withRouter.js | 957 kB | 957 kB | ✓ |
| Overall change | 4.72 MB | 4.72 MB | ✓ |

</details>

</details>
Commit: 965cf682ec76fe05ed622932f166673cb117a597

**Pull request merged**